airbyte-integrations/connectors/source-open-exchange-rates/source_open_exchange_rates/source.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import pendulum
+from pendulum import DateTime
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+class OpenExchangeRates(HttpStream, ABC):
+    url_base = "https://openexchangerates.org/api/"
+
+    primary_key = None
+    cursor_field = "date"
+
+
+    def __init__(self, base: Optional[str], start_date: str, app_id: str, **kwargs: dict) -> None:
+        super().__init__(**kwargs)
+
+        self.base = base
+        self.start_date = start_date
+        self.app_id = app_id
+        self._cursor_value = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {}
+
+        if self.base is not None:
+            params["base"] = self.base
+
+        return params
+
+    def request_headers(self, **kwargs) -> MutableMapping[str, Any]:
+        return self.authenticator.get_auth_header()
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if self._cursor_value:
+            return {self.cursor_field: self._cursor_value}
+        else:
+            return {self.cursor_field: self.start_date}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = pendulum.parse(value[self.cursor_field])
+
+    def parse_response(self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any] = None,
+        stream_slice: Mapping[str, Any] = None
+    ) -> Iterable[Mapping]:
```

Comment:
Can we make the signature of this match the one on the abstract class?

```suggestion
    def parse_response(
        self,
        response: requests.Response,
        *,
        stream_state: Mapping[str, Any],
        stream_slice: Mapping[str, Any] = None,
        next_page_token: Mapping[str, Any] = None,
    ) -> Iterable[Mapping]:
```